### PR TITLE
Ignore any canvas move events that are outside of the canvas borders

### PR DESCRIPTION
**Pull request by halatmit**

### CODE REVIEW

**Comment by ewpatton** [comment]
LGTM, but I wonder if we should have a flag on the Canvas just in case there are projects out there that rely on the fact that drag events can be reported outside of the canvas bounds.

**Comment by halatmit** [comment]
I did not generate a new companion here, or change the preferred companions in YaVersion.java
should I have done that?

**Comment by halatmit** [comment]
I didn't generate a new companion or change the acceptable companions in YaVersion.java.
Should I have done that?

**Comment by ewpatton** [comment]
@halatmit No, that's fine. When I test I build a fresh companion as part of the testing procedure and use that.

File: appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java [hunk 1]
```diff
@@ -278,6 +279,16 @@ void parse(MotionEvent event) {
           isDrag = true;
           drag = true;
 
+	  // Don't let MOVE extend beyond the bounds of the canvas
+	  // uf ExtendMovesOutsideCanvas is false
+	  if (
```
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java]: Google Java style guide would not have a line break between lines 284 and 285, and betwee lines 286-288.

**Pull request merged**